airflow/io/store/__init__.py

```diff
@@ -134,6 +146,6 @@ def attach(
         if store := _STORE_CACHE.get(alias):
             return store
 
-    _STORE_CACHE[alias] = store = ObjectStore(protocol=protocol, conn_id=conn_id, fs=fs)
+    _STORE_CACHE[alias] = store = ObjectStore(protocol=protocol, conn_id=conn_id, fs=fs, **kwargs)
```

Comment:
It makes me uncomfortable that storage options are only effective in the first call to `attach`; in later calls they will be silently ignored, even when options in the existing storage object does not match. It would be more explicit (thus less prone to user errors) if storage creation and retrieval should be splitted into two functions instead.

Reply:
Yeah, you have a point. Do you have a suggestion?